### PR TITLE
Bluetooth: Support `BT_RECV_WORKQ_BT` with Zephyr LL

### PR DESCRIPTION
**Pull request by alwa-nordic**
This patch allows selecting the combination of `CONFIG_BT_RECV_WORKQ_BT` and `BT_LL_SW_SPLIT`. The implementation of `bt_recv_prio` is copied from `hci_raw.c`. This ensures the order of packets is the same as when the controller's `hci_driver.c` is off-chip and sends HCI over UART.

Without this patch, the Zephyr LL driver will duplicate some events on behalf of `hci_core`. This duplication is not wanted when `CONFIG_BT_RECV_WORKQ_BT=y`.

The above mentioned duplication is not sound and is evidenced in logged warnings, e.g. in `tests/bsim/bluetooth/host/l2cap/stress` with `CONFIG_BT_RECV_WORKQ_BT=y` logging "no transition".

This patch is a step towards the deprecation of `CONFIG_BT_RECV_BLOCKING`. `CONFIG_BT_RECV_BLOCKING` is complicated and tightly coupled to `hci_core`. In the future, removing `CONFIG_BT_RECV_BLOCKING` will give a cleaner separation between `hci_core` and the drivers and allow `hci_core` to evolve without changes spilling out into the drivers.

### CODE REVIEW

**Review by jori-nordic — DISMISSED** [review]

**Comment by alwa-nordic** [comment]
Updated comment and commit message.

**Review by cvinayak — CHANGES_REQUESTED** [review]
Is there a `tests/bluetooth/init/` build that covers `CONFIG_BT_RECV_BLOCKING=n` and `CONFIG_BT_RECV_BLOCKING=y` ?

**Comment by jori-nordic** [comment]
> Is there a `tests/bluetooth/init/` build that covers `CONFIG_BT_RECV_BLOCKING=n` and `CONFIG_BT_RECV_BLOCKING=y` ?

> This patch is a step towards the deprecation of `CONFIG_BT_RECV_BLOCKING`

It's going away very soon.

**Comment by alwa-nordic** [comment]
> Is there a `tests/bluetooth/init/` build that covers `CONFIG_BT_RECV_BLOCKING=n` and `CONFIG_BT_RECV_BLOCKING=y` ?

I added an entry to `tests/bluetooth/init/testcase.yaml`.

**Review by jori-nordic — DISMISSED** [review]

**Comment by cvinayak** [comment]
> > Is there a `tests/bluetooth/init/` build that covers `CONFIG_BT_RECV_BLOCKING=n` and `CONFIG_BT_RECV_BLOCKING=y` ?
> 
> I added an entry to `tests/bluetooth/init/testcase.yaml`.

Latest head on the PR does not have it, your force-push removed it...

File: tests/bluetooth/init/testcase.yaml [hunk 1]
```diff
@@ -341,3 +341,7 @@ tests:
       - nrf52840dk_nrf52840
       - nrf52dk_nrf52832
       - rv32m1_vega_ri5cy
+  bluetooth.init.test_config_bt_recv_workq_bt:
+    extra_args:
+      - CONFIG_BT_RECV_WORKQ_BT=y
+    platform_allow: qemu_cortex_m3
```
Comment (cvinayak) [inline tests/bluetooth/init/testcase.yaml:347]: Why QEMU and not nRF5x SoC?
```suggestion
    platform_allow:
      - nrf52840dk_nrf52840
```

File: tests/bluetooth/init/testcase.yaml [hunk 2]
```diff
@@ -341,3 +341,7 @@ tests:
       - nrf52840dk_nrf52840
       - nrf52dk_nrf52832
       - rv32m1_vega_ri5cy
+  bluetooth.init.test_config_bt_recv_workq_bt:
+    extra_args:
```
Comment (cvinayak) [inline tests/bluetooth/init/testcase.yaml:345]:
```suggestion
  bluetooth.init.test_config_bt_recv_workq_bt:
    extra_args:
      - CONF_FILE=prj_ctlr.conf
```

**Review by cvinayak — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by aescolar** [comment]
@alwa-nordic  @cvinayak Should this target 3.7 (given that we are after rc1 for 3.6) or is this considered a bug?

**Comment by alwa-nordic** [comment]
> @alwa-nordic @cvinayak Should this target 3.7 (given that we are after rc1 for 3.6) or is this considered a bug?

I think we don't need to treat it as a bug. We have not had a bug report for this from any user. It is possible to select a `CONFIG_BT_RECV_CONTEXT` kconfig that will result in a bug, but I doubt anyone would do this. The `CONFIG_BT_RECV_CONTEXT` has defaults that are ok, and there is no reason for users to change it.

**Pull request merged**